packages/vite/src/node/cli.ts

```diff
@@ -235,11 +237,12 @@ cli
         https?: boolean
         open?: boolean | string
         strictPort?: boolean
+        outDir?: string
       } & GlobalCLIOptions
     ) => {
       const { preview } = await import('./preview')
       try {
-        const server = await preview({
+        const inlineConfig: InlineConfig = {
           root,
           base: options.base,
           configFile: options.config,
```

Comment:
Are there any reasons not to write `build.outDir` directly here?

Reply:
[`InlineConfig` is merged with the config file by `mergeConfig`](https://github.com/vitejs/vite/blob/dd59f7e82098057895d62b35092175ae5f5bb2b8/packages/vite/src/node/config.ts#L400) and that [skips `undefined`](https://github.com/vitejs/vite/blob/dd59f7e82098057895d62b35092175ae5f5bb2b8/packages/vite/src/node/utils.ts#L1045-L1047).
So I think it works when `outDir: undefined`.

Reply:
Oh that's really strange. Your suggestion worked! Last I implemented it doesn't. Updating this one.